### DISCUSSION TITLE
Swap-in transaction principle

### DISCUSSION THREAD

**Discussion by margre8** [ACINQ/phoenix#399] [Q&A] [answered]
Dear Phoenix developers,

I would like to ask You about new Swap-in transactions (Phoenix 2.0). Am I understand correctly that the Phoenix BTC-recieve Address is a multisig address which is directly used to open a Payment Channel? If so what happens when Someone sends too few funds, that Channel cannot be opened?

**Comment by dpad85** [comment, marked as answer]
Sorry for not replying sooner. 

> Address is a multisig address which is directly used to open a Payment Channel

Yes, the multisig scheme allows us to make the swap 0-conf (see [swap-in potentiam](https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-January/003810.html))

> what happens when Someone sends too few funds

Too few funds actually means : the cost of the swap would be too much relative to the amount to be swapped. By default, Phoenix will prevent the cost to be over 5000 sat / 50% of the amount received (that's configurable). So it all depends on the current on-chain feerate.

So if that happens, the funds remain on the address, until the feerate decrease enough that the swap can be made - or until more funds are deposited to this address and make the fee affordable.
